ui/src/core_components/content/CoreLink.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+	<div class="CoreLink">
+		<a
+			:href="fields.url.value"
+			:target="fields.target.value"
+			:rel="fields.rel.value"
+		>
+			{{ displayText }}
+		</a>
+	</div>
+</template>
+
+<script lang="ts">
+import { FieldType, Core } from "../../streamsyncTypes";
+import { cssClasses, primaryTextColor } from "../../renderer/sharedStyleFields";
+import injectionKeys from "../../injectionKeys";
+export default {
+	streamsync: {
+		name: "Link",
+		description: "A component to create a hyperlink.",
+		category: "Content",
+		fields: {
+			url: {
+				name: "URL",
+				type: FieldType.Text,
+				desc: "A valid URL.",
+				options: (ss: Core) => {
+					return Object.fromEntries(
+						ss
+							.getComponents("root", true)
+							.map((page) => page.content.key)
```

Comment:
Careful with this one. Like I was saying earlier not all pages will have keys. In fact I expect most pages to not have keys.

Reply:
Ah sorry didn't see the next line

Reply:
thats why there is filter in the next line